tests/integration/test_ingress_relation.py

```diff
@@ -74,3 +74,67 @@ def _on_ingress_relation_changed(self, event):
     response = requests.get("http://127.0.0.1/path/ok", headers={"Host": "any"}, timeout=5)
     assert response.text == "http://any/path"
     assert response.status_code == 200
+
+
+async def test_ingress_relation_strip_prefix(
+    model: Model, deploy_any_charm, run_action, build_and_deploy_ingress
+):
+    """
+    arrange: deploy ingress integrator and any-charm with strip_prefix=True.
+    action: configure path-routes and request URL through ingress.
+    assert: /path/ok is rewritten to /ok at backend.
+    """
+    any_charm_py = textwrap.dedent(
+        """\
+    import pathlib
+    import subprocess
+    import ops
+    from any_charm_base import AnyCharmBase
+    from ingress import IngressPerAppRequirer
+    class AnyCharm(AnyCharmBase):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            # require ingress relation with strip_prefix=True
+            self.ingress = IngressPerAppRequirer(self, port=8080, strip_prefix=True)
+            self.unit.status = ops.BlockedStatus("Waiting for ingress relation")
+            self.framework.observe(
+                self.on.ingress_relation_changed, self._on_ingress_relation_changed
+            )
+
+        def start_server(self):
+            www_dir = pathlib.Path("/tmp/www")
+            www_dir.mkdir(exist_ok=True)
+            file_path = www_dir / "ok"
+            file_path.parent.mkdir(exist_ok=True)
+            file_path.write_text(str(self.ingress.url))
+            proc_http = subprocess.Popen(
+                ["python3", "-m", "http.server", "-d", www_dir, "8080"],
+                start_new_session=True,
+            )
+
+        def _on_ingress_relation_changed(self, event):
+            self.unit.status = ops.ActiveStatus()
+    """
+    )
+
+    src_overwrite = {
+        "ingress.py": pathlib.Path("lib/charms/traefik_k8s/v2/ingress.py").read_text(
+            encoding="utf-8"
+        ),
+        "any_charm.py": any_charm_py,
+    }
+
+    _, ingress = await asyncio.gather(
+        deploy_any_charm(json.dumps(src_overwrite)),
+        build_and_deploy_ingress(),
+    )
+
+    await ingress.set_config({"service-hostname": "any", "path-routes": "/path(/|$)(.*)"})
```

Comment:
The `path-routes` is part of the integration functionality that needs to be tested, so I think we shouldn't set it in the charm configuration during tests.